### DISCUSSION TITLE
Lists fail to properly support switching between different languages outside of UTF-8/Basic Multilingual Plane

### DISCUSSION THREAD

**Discussion by jculpon** [community/community#17102] [Other Feature Feedback, Questions, & Ideas]
This is a condensation of several misbehaviors I have observed using the beta lists features while attempting to communicate electronically in the following languages:
- Welsh/y Gymraeg/Cymragh/Brettonic Welsh
- 日本語
- 中国語・広東語
- Vietnamese, with or without French-associated diacritics
- French (FR_fr/AZERTY)
- QWERTY-based French (CSA_FR_fr/QWERTY)
- Lang Francoise Cajun Creole du nord/Louisiana Cajun
- Sanskrit-based languages
- Mathematical symbols used in typsetting LaTeX and Markdown documents

Based on my past experience with serious errors in transcoding between UTF representations and ShiftJIS vs ASCII vs EBDIC, I believe this accessibility issue may trigger photosensitivity issues in users with the following conditions:
- photosenstive epilepsy (see CDProjekt Red's Cybperpunk title which was pulled from Sony Computer Entertainment's PlayStation Network due to a launch bug involving unlicensed use of a medical test pattern without appropriate seizure protections)
- Post-traumatic Stress Disorders related to medical/physical trauma. (Citations available upon request)
- Photo-sensitive phases of predrome to postdrome Migraine headache. (Citations available upon request)

Releasing even a beta version of this feature to developers represents, in my opinion, a potential attack vector for individuals with hearing, vision, or other assistive needs to be have their serious medical conditions invoked by other individuals viewing or reviewing lists and code while "pair programming" or engaging in document reviews.

As of 2022/05/22 I am unaware of any practical examples of this attack being used against developers outside of the GitHub public development forums. However, due to the complexity of the issues involve, I STRONGLY SUGGEST (See Internet RFC [7322](7322:https://www.rfc-editor.org/rfc/rfc7322)) that this issue be marked private and SECURITY relevant before USENIX PEPR 2022 or USENIX/OSDI 2022.

The above text is released to GitHub.com as feedback for research and evaluation purposes ONLY, without any warranty of any kind. In case of confusion, I grant GitHub.com a one-time, revocable license to review this text. All other words on this thread belong to their creators unless otherwise prohibited by the Creative Commons BY-SA-NC >4.0a License or the latest version of the NetBSD license as appears on http://www.netbsd.org

**Comment by jculpon** [comment]
As I have not yet reviewed the animation code or software being used by GitHub.com, I also imply but do not assert or affirm the problem:
- These medical conditions are real
- Other developers do have them, and it may be unlawful under a strict interpretation of local statutes to distribute code that causes these kinds of injuries without the approval of local authorities
- I am currently residing in the United States of America
- The Chatham House Rule remains in effect.

Thank you/有難うございます！

**Reply by metinoy** [reply]
> As I have not yet reviewed the animation code or software being used by GitHub.com, I also imply but do not assert or affirm the problem:
> 
> * These medical conditions are real
> * Other developers do have them, and it may be unlawful under a strict interpretation of local statutes to distribute code that causes these kinds of injuries without the approval of local authorities
> * I am currently residing in the United States of America
> * The Chatham House Rule remains in effect.
> 
> Thank you/有難うございます！

**Comment by benjiallen** [comment]
@jculpon thanks for the feedback. I need a little help reproducing this issue. Can you upload a screenshot or provide a more detailed explanation of which feature you are referring to when you say "beta lists feature"? We have a few features which might fit the description of "lists".

My first thought was that you were referring to [task lists](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists). Is that assumption correct?

**Reply by jculpon** [reply]
My apologies, I'm currently ill so I'm not 100% sure. I believe this was on the all the community projects beta list interfaces, not just a specific tasking list. The issue was caused by (in my guesstimate) weird strobing in how sortin gand filtering were happening, but it's been a few weeks and I'm sick so no great guarantees and I don't have the energy currently to track it down again.

**Reply by jculpon** [reply]
Sorry for the delays in my response here; I've been bouncing between free/open source work and tech conferences. I was referring to any list interface that offers autocompletion advice on new/beta community or projects.

**Reply by jculpon** [reply]
https://github.com/jculpon/alpha-beta-pruning/issues/3 has an example that's f/oss relevant to PROJECTS beta but I see the same behavior in COMMUNITY.
